## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a1c10521-6b7c-4e9e-a36d-9d1a85864994) |
| mint-dist-erasure.sh | :x: [more...](#09e006b7-d899-488c-a902-a5083056937b) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#cebb9f56-c663-41b3-9e87-6cf82a8149b4) |
| mint-pools.sh | :x: [more...](#b6bb2e0a-5d9b-4ced-a60b-f78a14f632ae) |
<a name="cebb9f56-c663-41b3-9e87-6cf82a8149b4"></a>
### 16416-f349070/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.1:31897
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp d200b2de25d5:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ogsbijdw1i900q",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364543,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C38C16D557C, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ogsbijdw1i900q",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364543,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C38C16D557C, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-ogsbijdw1i900q",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 364543,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C38C16D557C, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2181.29,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 11 seconds
{
  "name": "awscli",
  "duration": 9182,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 0 seconds
(2/14) Running mc tests ... FAILED in 0 seconds
(2/14) Running minio-go tests ... FAILED in 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-yhbpqz5qukket1v9",
    "region": "eu-west-1"
  },
  "duration": 4287,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 192,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-h0qcpo, objectName = null, resource = /minio-java-test-h0qcpo, requestId = 173A8C3EE10B265C, hostId = null)\nrequest={method=PUT, url=http://15.15.15.1:31897/minio-java-test-h0qcpo, headers=Host: 15.15.15.1:31897\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.6\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20230115T174013Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 300\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 173A8C3EE10B265C\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sun, 15 Jan 2023 17:40:13 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(2/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(2/14) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-43f5928c-ae59-4deb-9e59-88acf4a9824a",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.1', port=31897): Max retries exceeded with url: /minio-py-test-43f5928c-ae59-4deb-9e59-88acf4a9824a (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.1', port=31897): Max retries exceeded with url: /minio-py-test-43f5928c-ae59-4deb-9e59-88acf4a9824a (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600540
}
(2/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45371",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (SlowDown): Resource requested is unreadable, please reduce your request rate"
}
(2/14) Running s3select tests ... FAILED in 2 minutes and 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-2e87f4a2-ea3d-49c1-a1b8-3f504e9e9ecd"
  },
  "duration": 126123,
  "message": "HTTPConnectionPool(host='15.15.15.1', port=31897): Max retries exceeded with url: /s3select-test-2e87f4a2-ea3d-49c1-a1b8-3f504e9e9ecd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2702e943a0>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f2702e943a0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.1', port=31897): Max retries exceeded with url: /s3select-test-2e87f4a2-ea3d-49c1-a1b8-3f504e9e9ecd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f2702e943a0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-r2gto3ciurzdks"
  },
  "duration": 313,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.1:31897/versioning-test-r2gto3ciurzdks\": dial tcp 15.15.15.1:31897: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
<a name="a1c10521-6b7c-4e9e-a36d-9d1a85864994"></a>
### 16416-f349070/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.6:30525
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp 68872c71a3ae:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-hj6epfg5ck0guh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366283,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3934A188A2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 4 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-hj6epfg5ck0guh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366283,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3934A188A2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-hj6epfg5ck0guh",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366283,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3934A188A2, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2190.79,
  "error": "We encountered an internal error, please try again.: cause(Read failed. Insufficient number of drives online)",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 11 seconds
{
  "name": "awscli",
  "duration": 9508,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 0 seconds
(2/14) Running mc tests ... FAILED in 1 seconds
(2/14) Running minio-go tests ... FAILED in 3 seconds
{
  "args": {
    "bucketName": "minio-go-test-hjfbtcps0m4t4v2e",
    "region": "eu-west-1"
  },
  "duration": 3244,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 188,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-83n4tn, objectName = null, resource = /minio-java-test-83n4tn, requestId = 173A8C3F9C2AC50C, hostId = null)\nrequest={method=PUT, url=http://15.15.15.6:30525/minio-java-test-83n4tn, headers=Host: 15.15.15.6:30525\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.6\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20230115T174016Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 300\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 173A8C3F9C2AC50C\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sun, 15 Jan 2023 17:40:16 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(2/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(2/14) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-6e1b7774-5d89-460c-9fe3-9ab837e42396",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.6', port=30525): Max retries exceeded with url: /minio-py-test-6e1b7774-5d89-460c-9fe3-9ab837e42396 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.6', port=30525): Max retries exceeded with url: /minio-py-test-6e1b7774-5d89-460c-9fe3-9ab837e42396 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600545
}
(2/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45353",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(2/14) Running s3select tests ... FAILED in 2 minutes and 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-1a2b95ce-0dd2-4f6a-9bf8-2f5b72381a5e"
  },
  "duration": 126124,
  "message": "HTTPConnectionPool(host='15.15.15.6', port=30525): Max retries exceeded with url: /s3select-test-1a2b95ce-0dd2-4f6a-9bf8-2f5b72381a5e (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f0545b07070>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f0545b07070>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.6', port=30525): Max retries exceeded with url: /s3select-test-1a2b95ce-0dd2-4f6a-9bf8-2f5b72381a5e (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f0545b07070>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-1kg2ptbnsbjxf6"
  },
  "duration": 291,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.6:30525/versioning-test-1kg2ptbnsbjxf6\": dial tcp 15.15.15.6:30525: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
<a name="b6bb2e0a-5d9b-4ced-a60b-f78a14f632ae"></a>
### 16416-f349070/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.2:32749
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp e99825056526:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-kk9693i09ux5ei",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366229,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3931263317, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 2 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-kk9693i09ux5ei",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366229,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3931263317, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-kk9693i09ux5ei",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366229,
  "error": "XMinioServerNotInitialized: Server not initialized, please try again.\n\tstatus code: 503, request id: 173A8C3931263317, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2173.46,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 9 seconds
{
  "name": "awscli",
  "duration": 7338,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 0 seconds
(2/14) Running mc tests ... FAILED in 0 seconds
(2/14) Running minio-go tests ... FAILED in 5 seconds
{
  "args": {
    "bucketName": "minio-go-test-lvdyl5ev2lnngbc3",
    "region": "eu-west-1"
  },
  "duration": 4945,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 186,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-1hsehd8, objectName = null, resource = /minio-java-test-1hsehd8, requestId = 173A8C3EF787F7D0, hostId = f0d10a60-56f3-4828-9813-9e3c3f89634b)\nrequest={method=PUT, url=http://15.15.15.2:32749/minio-java-test-1hsehd8, headers=Host: 15.15.15.2:32749\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.6\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20230115T174013Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 338\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 173A8C3EF787F7D0\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sun, 15 Jan 2023 17:40:13 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(2/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 4,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(2/14) Running minio-py tests ... FAILED in 10 minutes and 0 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-91113576-cab1-4fdc-b387-42c4bd5cd568",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.2', port=32749): Max retries exceeded with url: /minio-py-test-91113576-cab1-4fdc-b387-42c4bd5cd568 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.2', port=32749): Max retries exceeded with url: /minio-py-test-91113576-cab1-4fdc-b387-42c4bd5cd568 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600429
}
(2/14) Running s3cmd tests ... FAILED in 46 seconds
{
  "name": "s3cmd",
  "duration": "45375",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(2/14) Running s3select tests ... FAILED in 2 minutes and 6 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-89cc6c7b-b274-4ad6-b93d-3c2189a6b1cd"
  },
  "duration": 126122,
  "message": "HTTPConnectionPool(host='15.15.15.2', port=32749): Max retries exceeded with url: /s3select-test-89cc6c7b-b274-4ad6-b93d-3c2189a6b1cd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f47f1dd8370>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f47f1dd8370>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.2', port=32749): Max retries exceeded with url: /s3select-test-89cc6c7b-b274-4ad6-b93d-3c2189a6b1cd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f47f1dd8370>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-czdcjmvf9gg0zs"
  },
  "duration": 372,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.2:32749/versioning-test-czdcjmvf9gg0zs\": dial tcp 15.15.15.2:32749: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
<a name="09e006b7-d899-488c-a902-a5083056937b"></a>
### 16416-f349070/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      15.15.15.5:31807
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp e629cd3719fb:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 6 minutes and 6 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6mfg0tvst05xrq",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366836,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C393AB1138E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-java tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6mfg0tvst05xrq",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366836,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C393AB1138E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-php tests ... FAILED in 1 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-6mfg0tvst05xrq",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 366836,
  "error": "SlowDown: Resource requested is unreadable, please reduce your request rate\n\tstatus code: 503, request id: 173A8C393AB1138E, host id: ",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}
(1/14) Running aws-sdk-ruby tests ... FAILED in 7 seconds
{
  "name": "aws-sdk-ruby",
  "function": "listBuckets",
  "args": {},
  "duration": 2171.14,
  "error": "Server not initialized, please try again.",
  "status": "FAIL"
}
(1/14) Running awscli tests ... FAILED in 4 seconds
{
  "name": "awscli",
  "duration": 2582,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (XMinioServerNotInitialized) when calling the CreateBucket operation (reached max retries: 4): Server not initialized, please try again."
}
(1/14) Running healthcheck tests ... done in 0 seconds
(2/14) Running mc tests ... FAILED in 1 seconds
(2/14) Running minio-go tests ... FAILED in 4 seconds
{
  "args": {
    "bucketName": "minio-go-test-p0vvas3s1y3pzm5j",
    "region": "eu-west-1"
  },
  "duration": 4789,
  "error": "Server not initialized, please try again.",
  "function": "MakeBucket(bucketName, region)",
  "message": "MakeBucket failed",
  "name": "minio-go: testMakeBucketErrorV2",
  "status": "FAIL"
}
(2/14) Running minio-java tests ... FAILED in 1 seconds
{
  "name": "minio-java",
  "function": "makeBucket()",
  "args": "[basic check]",
  "duration": 218,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = XMinioServerNotInitialized, message = Server not initialized, please try again., bucketName = minio-java-test-1qulvt0, objectName = null, resource = /minio-java-test-1qulvt0, requestId = 173A8C3DEB7AB3A5, hostId = null)\nrequest={method=PUT, url=http://15.15.15.5:31807/minio-java-test-1qulvt0, headers=Host: 15.15.15.5:31807\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.6\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20230115T174009Z\nAuthorization: ██\n}\nresponse={code=503, headers=Accept-Ranges: bytes\nContent-Length: 302\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nRetry-After: 120\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 173A8C3DEB7AB3A5\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Sun, 15 Jan 2023 17:40:09 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(2/14) Running minio-js tests ... FAILED in 1 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 5,
  "status": "FAIL",
  "error": "S3Error: Server not initialized, please try again. at Object.parseError (node_modules/minio/dist/main/xml-parsers.js:71:11) at /mint/run/core/minio-js/node_modules/minio/dist/main/transformers.js:166:22 at DestroyableTransform._flush (node_modules/minio/dist/main/transformers.js:90:10) at DestroyableTransform.prefinish (node_modules/readable-stream/lib/_stream_transform.js:129:10) at prefinish (node_modules/readable-stream/lib/_stream_writable.js:611:14) at finishMaybe (node_modules/readable-stream/lib/_stream_writable.js:620:5) at endWritable (node_modules/readable-stream/lib/_stream_writable.js:643:3) at DestroyableTransform.Writable.end (node_modules/readable-stream/lib/_stream_writable.js:571:22) at IncomingMessage.onend (internal/streams/readable.js:670:10) at endReadableNT (internal/streams/readable.js:1333:12) at processTicksAndRejections (internal/process/task_queues.js:82:21)"
}
(2/14) Running minio-py tests ... FAILED in 10 minutes and 1 seconds
{
  "name": "minio-py:test_make_bucket_default_region",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-73c66c44-f41a-44cb-b01f-35b455ef0c21",
    "location": "default value ('us-east-1')"
  },
  "message": "HTTPConnectionPool(host='15.15.15.5', port=31807): Max retries exceeded with url: /minio-py-test-73c66c44-f41a-44cb-b01f-35b455ef0c21 (Caused by ResponseError('too many 503 error responses'))",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 168, in test_make_bucket_default_region\n    _call(log_entry, _CLIENT.make_bucket, bucket_name)\n  File \"/mint/run/core/minio-py/tests.py\", line 109, in _call\n    return func(*args, **kwargs)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  [Previous line repeated 2 more times]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 868, in urlopen\n    retries = retries.increment(method, url, response=response, _pool=self)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.5', port=31807): Max retries exceeded with url: /minio-py-test-73c66c44-f41a-44cb-b01f-35b455ef0c21 (Caused by ResponseError('too many 503 error responses'))\n",
  "function": "make_bucket(bucket_name, location, object_lock)",
  "duration": 600549
}
(2/14) Running s3cmd tests ... FAILED in 45 seconds
{
  "name": "s3cmd",
  "duration": "45374",
  "function": "test_make_bucket",
  "status": "FAIL",
  "error": "WARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 3 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 6 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 9 sec...\nWARNING: Retrying failed request: / (503 (XMinioServerNotInitialized): Server not initialized, please try again.)\nWARNING: Waiting 12 sec...\nWARNING: Retrying failed request: / (503 (SlowDown): Resource requested is unreadable, please reduce your request rate)\nWARNING: Waiting 15 sec...\nERROR: S3 error: 503 (XMinioServerNotInitialized): Server not initialized, please try again."
}
(2/14) Running s3select tests ... FAILED in 2 minutes and 7 seconds
{
  "name": "s3select:test_csv_input_quote_char",
  "function": "select_object_content(bucket_name, object_name, request)",
  "args": {
    "bucket_name": "s3select-test-bb339c48-24c6-42dc-80a2-25b98db7c6fd"
  },
  "duration": 126129,
  "message": "HTTPConnectionPool(host='15.15.15.5', port=31807): Max retries exceeded with url: /s3select-test-bb339c48-24c6-42dc-80a2-25b98db7c6fd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83bc9333a0>: Failed to establish a new connection: [Errno 111] Connection refused'))",
  "error": "Traceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 174, in _new_conn\n    conn = connection.create_connection(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 95, in create_connection\n    raise err\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/connection.py\", line 85, in create_connection\n    sock.connect(sa)\nConnectionRefusedError: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 703, in urlopen\n    httplib_response = self._make_request(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 398, in _make_request\n    conn.request(method, url, **httplib_request_kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 239, in request\n    super(HTTPConnection, self).request(method, url, body=body, headers=headers)\n  File \"/usr/lib/python3.8/http/client.py\", line 1256, in request\n    self._send_request(method, url, body, headers, encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1302, in _send_request\n    self.endheaders(body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1251, in endheaders\n    self._send_output(message_body, encode_chunked=encode_chunked)\n  File \"/usr/lib/python3.8/http/client.py\", line 1011, in _send_output\n    self.send(msg)\n  File \"/usr/lib/python3.8/http/client.py\", line 951, in send\n    self.connect()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 205, in connect\n    conn = self._new_conn()\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connection.py\", line 186, in _new_conn\n    raise NewConnectionError(\nurllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7f83bc9333a0>: Failed to establish a new connection: [Errno 111] Connection refused\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"./tests.py\", line 53, in main\n    test_csv_input_custom_quote_char(client, log_output)\n  File \"/mint/run/core/s3select/csv.py\", line 95, in test_csv_input_custom_quote_char\n    client.make_bucket(bucket_name)\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 618, in make_bucket\n    self._url_open(\n  File \"/usr/local/lib/python3.8/dist-packages/minio/api.py\", line 266, in _url_open\n    response = self._http.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 878, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  [Previous line repeated 1 more time]\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/usr/local/lib/python3.8/dist-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\nurllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='15.15.15.5', port=31807): Max retries exceeded with url: /s3select-test-bb339c48-24c6-42dc-80a2-25b98db7c6fd (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f83bc9333a0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n",
  "status": "FAIL"
}
(2/14) Running versioning tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "versioning-test-bymtr63axgaoha"
  },
  "duration": 311,
  "error": "RequestError: send request failed\ncaused by: Put \"http://15.15.15.5:31807/versioning-test-bymtr63axgaoha\": dial tcp 15.15.15.5:31807: connect: connection refused",
  "function": "/mint/build/versioning/bucket.go:47: main.testMakeBucket",
  "message": "Versioning CreateBucket Failed",
  "name": "versioning",
  "status": "FAIL"
}

Executed 1 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/admin-handlers-pools.go
+++ b/cmd/admin-handlers-pools.go
@@ -22,6 +22,7 @@ import (
 	"errors"
 	"fmt"
 	"net/http"
+	"strings"
 
 	"github.com/gorilla/mux"
 	"github.com/minio/minio/internal/logger"
@@ -49,28 +50,53 @@ func (a adminAPIHandlers) StartDecommission(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	pools, ok := objectAPI.(*erasureServerPools)
-	if !ok {
+	z, ok := objectAPI.(*erasureServerPools)
+	if !ok || len(z.serverPools) == 1 {
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrNotImplemented), r.URL)
 		return
 	}
 
-	if pools.IsRebalanceStarted() {
-		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, errDecommissionRebalanceAlreadyRunning), r.URL)
+	if z.IsDecommissionRunning() {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, errDecommissionAlreadyRunning), r.URL)
+		return
+	}
+
+	if z.IsRebalanceStarted() {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrAdminRebalanceAlreadyStarted), r.URL)
 		return
 	}
 
 	vars := mux.Vars(r)
 	v := vars["pool"]
 
-	idx := globalEndpoints.GetPoolIdx(v)
-	if idx == -1 {
-		// We didn't find any matching pools, invalid input
-		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, errInvalidArgument), r.URL)
-		return
+	pools := strings.Split(v, ",")
+	poolIndices := make([]int, 0, len(pools))
+
+	for _, pool := range pools {
+		idx := globalEndpoints.GetPoolIdx(pool)
+		if idx == -1 {
+			// We didn't find any matching pools, invalid input
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, errInvalidArgument), r.URL)
+			return
+		}
+		var pool *erasureSets
+		for pidx := range z.serverPools {
+			if pidx == idx {
+				pool = z.serverPools[idx]
+				break
+			}
+		}
+		if pool == nil {
+			// We didn't find any matching pools, invalid input
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, errInvalidArgument), r.URL)
+			return
+		}
+
+		poolIndices = append(poolIndices, idx)
 	}
 
-	if ep := globalEndpoints[idx].Endpoints[0]; !ep.IsLocal {
+	if len(poolIndices) > 0 && globalEndpoints[poolIndices[0]].Endpoints[0].IsLocal {
+		ep := globalEndpoints[poolIndices[0]].Endpoints[0]
 		for nodeIdx, proxyEp := range globalProxyEndpoints {
 			if proxyEp.Endpoint.Host == ep.Host {
 				if proxyRequestByNodeIndex(ctx, w, r, nodeIdx) {
@@ -80,7 +106,7 @@ func (a adminAPIHandlers) StartDecommission(w http.ResponseWriter, r *http.Reque
 		}
 	}
 
-	if err := pools.Decommission(r.Context(), idx); err != nil {
+	if err := z.Decommission(r.Context(), poolIndices...); err != nil {
 		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
 		return
 	}

--- a/cmd/erasure-server-pool-decom.go
+++ b/cmd/erasure-server-pool-decom.go
@@ -126,7 +126,7 @@ type poolMeta struct {
 
 // A decommission resumable tells us if decommission is worth
 // resuming upon restart of a cluster.
-func (p *poolMeta) returnResumablePools(n int) []PoolStatus {
+func (p *poolMeta) returnResumablePools() []PoolStatus {
 	var newPools []PoolStatus
 	for _, pool := range p.Pools {
 		if pool.Decommission == nil {
@@ -139,11 +139,8 @@ func (p *poolMeta) returnResumablePools(n int) []PoolStatus {
 			continue
 		} // In all other situations we need to resume
 		newPools = append(newPools, pool)
-		if n > 0 && len(newPools) == n {
-			return newPools
-		}
 	}
-	return nil
+	return newPools
 }
 
 func (p *poolMeta) DecommissionComplete(idx int) bool {
@@ -251,18 +248,6 @@ var (
 )
 
 func (p *poolMeta) Decommission(idx int, pi poolSpaceInfo) error {
-	for i, pool := range p.Pools {
-		if idx == i {
-			continue
-		}
-		if pool.Decommission != nil {
-			// Do not allow multiple decommissions at the same time.
-			// We shall for now only allow one pool decommission at
-			// a time.
-			return fmt.Errorf("%w at index: %d", errDecommissionAlreadyRunning, i)
-		}
-	}
-
 	// Return an error when there is decommission on going - the user needs
 	// to explicitly cancel it first in order to restart decommissioning again.
 	if p.Pools[idx].Decommission != nil &&
@@ -510,7 +495,6 @@ func (z *erasureServerPools) Init(ctx context.Context) error {
 	z.StartRebalance()
 
 	meta := poolMeta{}
-
 	if err := meta.load(ctx, z.serverPools[0], z.serverPools); err != nil {
 		return err
 	}
@@ -524,38 +508,38 @@ func (z *erasureServerPools) Init(ctx context.Context) error {
 	if !update {
 		z.poolMeta = meta
 
-		// We are only supporting single pool decommission at this time
-		// so it makes sense to only resume single pools at any given
-		// time, in future meta.returnResumablePools() might take
-		// '-1' as argument to decommission multiple pools at a time
-		// but this is not a priority at the moment.
-		for _, pool := range meta.returnResumablePools(1) {
+		pools := meta.returnResumablePools()
+		poolIndices := make([]int, 0, len(pools))
+		for _, pool := range pools {
 			idx := globalEndpoints.GetPoolIdx(pool.CmdLine)
 			if idx == -1 {
 				return fmt.Errorf("unexpected state present for decommission status pool(%s) not found", pool.CmdLine)
 			}
-			if globalEndpoints[idx].Endpoints[0].IsLocal {
-				go func(pool PoolStatus) {
-					r := rand.New(rand.NewSource(time.Now().UnixNano()))
-					for {
-						if err := z.Decommission(ctx, pool.ID); err != nil {
-							if errors.Is(err, errDecommissionAlreadyRunning) {
-								// A previous decommission running found restart it.
+			poolIndices = append(poolIndices, idx)
+		}
+
+		if len(poolIndices) > 0 && globalEndpoints[poolIndices[0]].Endpoints[0].IsLocal {
+			go func() {
+				r := rand.New(rand.NewSource(time.Now().UnixNano()))
+				for {
+					if err := z.Decommission(ctx, poolIndices...); err != nil {
+						if errors.Is(err, errDecommissionAlreadyRunning) {
+							// A previous decommission running found restart it.
+							for _, idx := range poolIndices {
 								z.doDecommissionInRoutine(ctx, idx)
-								return
 							}
-							if configRetriableErrors(err) {
-								logger.LogIf(ctx, fmt.Errorf("Unable to resume decommission of pool %v: %w: retrying..", pool, err))
-								time.Sleep(time.Second + time.Duration(r.Float64()*float64(5*time.Second)))
-								continue
-							}
-							logger.LogIf(ctx, fmt.Errorf("Unable to resume decommission of pool %v: %w", pool, err))
 							return
 						}
-						break
+						if configRetriableErrors(err) {
+							logger.LogIf(ctx, fmt.Errorf("Unable to resume decommission of pools %v: %w: retrying..", pools, err))
+							time.Sleep(time.Second + time.Duration(r.Float64()*float64(5*time.Second)))
+							continue
+						}
+						logger.LogIf(ctx, fmt.Errorf("Unable to resume decommission of pool %v: %w", pools, err))
+						return
 					}
-				}(pool)
-			}
+				}
+			}()
 		}
 
 		return nil
@@ -1039,11 +1023,12 @@ func (z *erasureServerPools) doDecommissionInRoutine(ctx context.Context, idx in
 	}
 
 	z.poolMetaMutex.Lock()
-	failed := z.poolMeta.Pools[idx].Decommission.ItemsDecommissionFailed > 0
+	failed := z.poolMeta.Pools[idx].Decommission.ItemsDecommissionFailed > 0 || contextCanceled(dctx)
+	poolCmdLine := z.poolMeta.Pools[idx].CmdLine
 	z.poolMetaMutex.Unlock()
 
 	if !failed {
-		logger.Info("Decommissioning almost complete - checking for left over objects")
+		logger.Info("Decommissioning complete for pool '%s', verifying for any pending objects", poolCmdLine)
 		err := z.checkAfterDecom(dctx, idx)
 		if err != nil {
 			logger.LogIf(ctx, err)
@@ -1067,8 +1052,8 @@ func (z *erasureServerPools) IsSuspended(idx int) bool {
 }
 
 // Decommission - start decommission session.
-func (z *erasureServerPools) Decommission(ctx context.Context, idx int) error {
-	if idx < 0 {
+func (z *erasureServerPools) Decommission(ctx context.Context, indices ...int) error {
+	if len(indices) == 0 {
 		return errInvalidArgument
 	}
 
@@ -1077,11 +1062,15 @@ func (z *erasureServerPools) Decommission(ctx context.Context, idx int) error {
 	}
 
 	// Make pool unwritable before decommissioning.
-	if err := z.StartDecommission(ctx, idx); err != nil {
+	if err := z.StartDecommission(ctx, indices...); err != nil {
 		return err
 	}
 
-	go z.doDecommissionInRoutine(ctx, idx)
+	go func() {
+		for _, idx := range indices {
+			z.doDecommissionInRoutine(ctx, idx)
+		}
+	}()
 
 	// Successfully started decommissioning.
 	return nil
@@ -1260,8 +1249,8 @@ func (z *erasureServerPools) getBucketsToDecommission(ctx context.Context) ([]de
 	return decomBuckets, nil
 }
 
-func (z *erasureServerPools) StartDecommission(ctx context.Context, idx int) (err error) {
-	if idx < 0 {
+func (z *erasureServerPools) StartDecommission(ctx context.Context, indices ...int) (err error) {
+	if len(indices) == 0 {
 		return errInvalidArgument
 	}
 
@@ -1304,34 +1293,28 @@ func (z *erasureServerPools) StartDecommission(ctx context.Context, idx int) (er
 		}
 	}
 
-	var pool *erasureSets
-	for pidx := range z.serverPools {
-		if pidx == idx {
-			pool = z.serverPools[idx]
-			break
-		}
-	}
-
-	if pool == nil {
-		return errInvalidArgument
-	}
-
-	pi, err := z.getDecommissionPoolSpaceInfo(idx)
-	if err != nil {
-		return err
-	}
-
 	z.poolMetaMutex.Lock()
 	defer z.poolMetaMutex.Unlock()
 
-	if err = z.poolMeta.Decommission(idx, pi); err != nil {
-		return err
+	for _, idx := range indices {
+		pi, err := z.getDecommissionPoolSpaceInfo(idx)
+		if err != nil {
+			return err
+		}
+
+		if err = z.poolMeta.Decommission(idx, pi); err != nil {
+			return err
+		}
+
+		z.poolMeta.QueueBuckets(idx, decomBuckets)
 	}
-	z.poolMeta.QueueBuckets(idx, decomBuckets)
+
 	if err = z.poolMeta.save(ctx, z.serverPools); err != nil {
 		return err
 	}
+
 	globalNotificationSys.ReloadPoolMeta(ctx)
+
 	return nil
 }
 

